dio/lib/src/failed_request_interceptor.dart

```diff
@@ -0,0 +1,21 @@
+// ignore_for_file: public_member_api_docs
+
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+
+class FailedRequestInterceptor extends Interceptor {
+  FailedRequestInterceptor({Hub? hub}) : _hub = hub ?? HubAdapter();
+
+  final Hub _hub;
+
+  @override
+  void onError(
+    DioError err,
+    ErrorInterceptorHandler handler,
+  ) {
+    _hub.captureException(err);
+    _hub.getSpan()?.throwable = err;
```

Comment:
```suggestion
    _hub.getSpan()?.throwable = err;
    _hub.captureException(err);
```

Reply:
What's the reason for calling one before the other?

Reply:
The event captured by `captureException` will contain the `event.contexts.trace` data from the span associated with the previously assigned `throwable`.

Reply:
Oh, yeah, in that case it should definitely be changed.